### PR TITLE
Check quota on explicit request/limits in pod templates

### PR DESCRIPTION
**Pull request by jwforres**
Partially implements https://trello.com/c/mfeqgKvL

### CODE REVIEW

**Comment by jwforres** [comment]
@spadgett PTAL, i'm not totally in love with the implementation, open to other ideas.  No matter what the idea is that we only check explicitly defined requests or limits, no defaulting behavior.  got agreement from @derekwaynecarr that this is the sanest approach

**Comment by spadgett** [comment]
I'm fine with the approach.

Are we thinking about checking `spec.replicas`? Or the total for everything you're creating? It looks like we only check the resources one by one.

**Comment by spadgett** [comment]
(I understand we need to draw the line somewhere.)

**Comment by jwforres** [comment]
Yeah I'm making that choice intentionally, we are only warning you that you wont be able to have *any* pods created.  If you can get at least one, then we won't warn you, because you'll get the warnings on the overview that you are out of resources.

**Comment by jwforres** [comment]
As far as checking the total across all pod template things you are trying to create, that gets even more complicated.  Since each quota may only apply to certain resources based on the scopes, we'd have to segment the totals by quota.

**Comment by spadgett** [comment]
> Since each quota may only apply to certain resources based on the scopes, we'd have to segment the totals by quota.

OK you've convinced me :)

**Comment by jwforres** [comment]
@spadgett ok pushed the dist files, any other comments or ready to merge ?

File: app/scripts/services/quota.js [hunk 1]
```diff
@@ -92,24 +92,66 @@ angular.module("openshiftConsole")
...
+              var detail;
+              if (resource.kind === 'Pod') {
+                detail = "You may not be able to create the " + humanizeKind(resource.kind) + " '" + resource.metadata.name + "'.";
+              }
+              else {
+                detail = "You can can still create " + humanizeKind(resource.kind) + " '" + resource.metadata.name + "' but you may not have pods created until resources are freed.";
+              }
+              alerts.push({
+                type: 'warning',
+                message: 'You are close to your quota for ' + humanizeQuotaResource(type).toLowerCase() + " on pods.",
```
Comment (spadgett) [inline app/scripts/services/quota.js]: Will this lowercase `CPU`? might want a case option on `humanizeQuotaResource` like we have for `humanizeKind`.

**Comment by spadgett** [comment]
Should `pods` be in the list of types we check? Looks like the logic assumes a compute resource.

**Comment by jwforres** [comment]
@spadgett updated

**Comment by spadgett** [comment]
[merge]

**Pull request merged**